File: Assets/GenerateMaps/Scripts/HUDElements.cs
using UnityEngine;
using System.Collections;

public class HUDElements : MonoBehaviour {
	
	public Texture2D bgBarTex;
	public Texture2D hpBarTex;
	public Texture2D expBarTex;

	private PlayerStats stats;

	private const float HP_BAR_X = 10;
	private const float HP_BAR_Y = 10;
	private float HP_BAR_WIDTH = 200; // Not a const because HP bar can become longer if player levels up
	private const float HP_BAR_HEIGHT = 20;

	private const float EXP_BAR_X = 10;
	private const float EXP_BAR_Y = 40;
	private const float EXP_BAR_WIDTH = 200;
	private const float EXP_BAR_HEIGHT = 20;

	private double hpBarPercent = 100.0;
	private double expBarPercent = 100.0;

	// Use this for initialization
	void Start () {
		stats = this.GetComponent<PlayerStats>();
	}
	
	// Update is called once per frame
	void Update () {
		hpBarPercent = (float)stats.GetCurHp() / (float)stats.GetMaxHp();
		expBarPercent = (float)stats.GetCurExp() / (float)stats.GetMaxExp();
	}

	void OnGUI () {
		GUI.DrawTexture(new Rect(HP_BAR_X, HP_BAR_Y, HP_BAR_WIDTH, HP_BAR_HEIGHT), bgBarTex);
		GUI.DrawTexture(new Rect(EXP_BAR_X, EXP_BAR_Y, EXP_BAR_WIDTH, EXP_BAR_HEIGHT), bgBarTex);
		if (stats.GetCurHp() > 0) {
			GUI.DrawTexture(new Rect(HP_BAR_X, HP_BAR_Y, HP_BAR_WIDTH * (float)hpBarPercent, HP_BAR_HEIGHT), hpBarTex);
		}
		
		if (stats.GetCurExp() > 0) {
			GUI.DrawTexture(new Rect(EXP_BAR_X, EXP_BAR_Y, EXP_BAR_WIDTH * (float)expBarPercent, EXP_BAR_HEIGHT), expBarTex);
		}
	}
}

File: Assets/GenerateMaps/Scripts/HUDElements.js
var player : GameObject;

var bgBar : Texture2D;
var healthBar : Texture2D;
var expBar : Texture2D;

var healthBarX : int = 10;
var healthBarY : int = 10;
var healthBarWidth : int = 200;
var healthBarHeight : int = 20;

var expBarX : int = 10;
var expBarY : int = 40;
var expBarWidth : int = 200;
var expBarHeight : int = 20;

private var healthBarPercent : float = 100.0;
private var expBarPercent : float = 100.0;


function Start () {
	
}

function Update () {
	healthBarPercent = player.GetComponent("PlayerStats").GetCurHealth() / player.GetComponent("PlayerStats").GetMaxHealth();
	expBarPercent = player.GetComponent("PlayerStats").GetCurExp() / player.GetComponent("PlayerStats").GetMaxExp();
	
	
}

function OnGUI () {
	GUI.DrawTexture(Rect(healthBarX, healthBarY, healthBarWidth, healthBarHeight), bgBar);
	GUI.DrawTexture(Rect(expBarX, expBarY, expBarWidth, expBarHeight), bgBar);
	if (player.GetComponent("PlayerStats").GetCurHealth() > 0) {
		GUI.DrawTexture(Rect(healthBarX, healthBarY, healthBarWidth * healthBarPercent, healthBarHeight), healthBar);
	}
	
	if (player.GetComponent("PlayerStats").GetCurExp() > 0) {
		GUI.DrawTexture(Rect(expBarX, expBarY, expBarWidth * expBarPercent, expBarHeight), expBar);
	}
}